### PR TITLE
[Projects] Fix file validation when set secrets or set envs 

### PR DESCRIPTION
**Pull request by GiladShapira94**
https://jira.iguazeng.com/browse/ML-3381

In addition, this PR includes a fix to the set_envs method.

### CODE REVIEW

**Review by alonmr — COMMENTED** [review]
Great job! minor comment

File: mlrun/runtimes/pod.py [hunk 1]
```diff
@@ -992,12 +992,14 @@ def set_envs(self, env_vars: dict = None, file_path: str = None):
                 "must specify env_vars OR file_path"
             )
         if file_path:
-            env_vars = dotenv.dotenv_values(file_path)
-            if None in env_vars.values():
-                raise mlrun.errors.MLRunInvalidArgumentError(
-                    "env file lines must be in the form key=value"
-                )
-
+            if os.path.exists(file_path):
```
[the remark this answers is not in the document]
Comment (GiladShapira94) [reply mlrun/runtimes/pod.py]: Thanks 👍  fixed

File: mlrun/projects/project.py [hunk 2]
```diff
@@ -1847,11 +1847,14 @@ def set_secrets(
...
-                    "env file lines must be in the form key=value"
-                )
+            if path.exists(file_path):
+                secrets = dotenv.dotenv_values(file_path)
+                if None in secrets.values():
+                    raise mlrun.errors.MLRunInvalidArgumentError(
+                        "env file lines must be in the form key=value"
+                    )
+            else:
+                raise mlrun.errors.MLRunNotFoundError(f"{file_path} does not exist.")
```
Comment (gtopper) [inline mlrun/projects/project.py:1857]:
```suggestion
                raise mlrun.errors.MLRunNotFoundError(f"{file_path} does not exist")
```
(it's the convention in the code base)

File: tests/runtimes/test_run.py [hunk 3]
```diff
@@ -312,3 +314,22 @@ def test_new_function_invalid_characters():
     invalid_function_name = "invalid_name with_spaces"
     function = mlrun.new_function(name=invalid_function_name, runtime=runtime)
     assert function.metadata.name == "invalid-name-with-spaces"
+
+
+def test_set_envs():
+    assets_path = pathlib.Path(pathlib.Path(__file__).parent).parent / "assets"
```
Comment (gtopper) [inline tests/runtimes/test_run.py:320]: Can be simplified?
```suggestion
    assets_path = pathlib.Path(__file__).parent.parent / "assets"
```

**Review by Tankilevitch — APPROVED** [review]
LGTM, just fix the suggested comments

**Review by Tankilevitch — APPROVED** [review]
LGTM, just fix the suggested comments

File: tests/runtimes/test_run.py (hunk 3 above)
Comment (GiladShapira94) [reply tests/runtimes/test_run.py]: Thanks, fixed 👍

**Pull request merged**